cmd/osbuild-composer/main.go

```diff
@@ -55,6 +59,18 @@ func createTLSConfig(c *connectionConfig) (*tls.Config, error) {
 		Certificates: []tls.Certificate{cert},
 		ClientAuth:   tls.RequireAndVerifyClientCert,
 		ClientCAs:    roots,
+		VerifyPeerCertificate: func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+			for _, chain := range verifiedChains {
+				for _, domainGlob := range c.DomainAllowlist {
+					commonName := chain[0].Subject.CommonName
```

Comment:
Should we match on SAN insetad? Isn't CN deprecated? @larskarlitski @gicmo ?

Reply:
It has been deprecated for > 10 years, but still works. @gicmo was seeing some warnings in python client libraries. Probably makes sense to switch or support both. I wouldn't block on it.

Reply:
This is the mentioned python library issue: https://github.com/urllib3/urllib3/issues/497, for background information.

Reply:
I used https://golang.org/pkg/crypto/x509/#Certificate.VerifyHostname in the end. Thanks, @gicmo, for the suggestion!